source/controllers/include/controllers/impedance/NewImpedance.hpp

```diff
@@ -0,0 +1,202 @@
+#pragma once
+
+#include <eigen3/Eigen/Core>
+#include <eigen3/Eigen/Dense>
+
+#include "controllers/IController.hpp"
+#include "state_representation/parameters/Parameter.hpp"
+#include "state_representation/State.hpp"
+
+namespace controllers::impedance {
+
+/**
+ * @class NewImpedance
+ * @brief Definition of an impedance controller in either joint or task space
+ * @tparam S the space of the controller (either CartesianState or JointState)
+ */
+template<class S>
+class NewImpedance : public IController<S> {
+public:
+
+  /**
+   * @brief Base constructor.
+   * @details This initializes all gain matrices to the identity matrix of the corresponding dimensionality.
+   * @param dimensions The number of dimensions associated with the controller
+   */
+  explicit NewImpedance(unsigned int dimensions = 6);
+
+  /**
+   * @brief Constructor from an initial parameter list
+   * @param parameters A parameter list containing initial gain values
+   * @param dimensions The number of dimensions associated with the controller
+   */
+  explicit NewImpedance(
+      const std::list<std::shared_ptr<state_representation::ParameterInterface>>& parameters,
+      unsigned int dimensions = 6
+  );
+
+  /**
+   * @copydoc IController::compute_command(const S&,const S&)
+   */
+  S compute_command(const S& command_state, const S& feedback_state) override;
+
+  /**
+   * @brief Getter of the damping matrix
+   * @return the damping matrix
+   */
+  [[nodiscard]] const Eigen::MatrixXd& get_damping() const;
```

Comment:
Do we really still want all those getters and setters? We removed them in dynamical systems...

Reply:
I removed them initially, but then I wanted to limit the amount of changes in one go. If I remove these (and other setters/getters in derived classes), then I have to change all references to these in the derived classes and tests. I first wanted to make everything factory-instantiable and test them in parallel. I guess now that I know it works (that the base pointer interface is constructed and behaves the same as the the direct object controller) I can remove these properly. 

It's just a design decision in the end, whether we want to allow direct object access to a controller in some cases or disable everything except the base interface. I would also tend towards exclusively using the consolidated base interface. @buschbapti what do you think?

Reply:
yes agreed on that. I don't think they are necessary anymore. The base access might be a little more verbose though but keeping them just adds overhead.